plugins/transport-grpc/src/main/java/org/opensearch/transport/grpc/ssl/SecureNetty4GrpcServerTransport.java

```diff
@@ -0,0 +1,97 @@
+import java.util.Locale;
+
+import io.grpc.BindableService;
+import io.grpc.netty.shaded.io.grpc.netty.NettyServerBuilder;
+import io.grpc.netty.shaded.io.netty.handler.ssl.ApplicationProtocolConfig;
+import io.grpc.netty.shaded.io.netty.handler.ssl.ApplicationProtocolNames;
+import io.grpc.netty.shaded.io.netty.handler.ssl.ClientAuth;
+import io.grpc.netty.shaded.io.netty.handler.ssl.JdkSslContext;
+import io.grpc.netty.shaded.io.netty.handler.ssl.SupportedCipherSuiteFilter;
+
+/**
+ * Netty4GrpcServerTransport with TLS enabled.
+ * Security settings injected through a SecureAuxTransportSettingsProvider.
+ */
+public class SecureNetty4GrpcServerTransport extends Netty4GrpcServerTransport {
+    private static final String[] DEFAULT_SSL_PROTOCOLS = { "TLSv1.3", "TLSv1.2", "TLSv1.1" };
+
+    /**
+     * Type key to select secure transport.
+     */
+    public static final String GRPC_SECURE_TRANSPORT_SETTING_KEY = "experimental-secure-transport-grpc";
+
+    /**
+     * Distinct port setting required as it depends on transport type key.
+     */
+    public static final Setting<PortsRange> SETTING_GRPC_SECURE_PORT = AUX_TRANSPORT_PORT.getConcreteSettingForNamespace(
+        GRPC_SECURE_TRANSPORT_SETTING_KEY
+    );
+
+    /**
+     * Creates a new SecureNetty4GrpcServerTransport instance and inject a SecureAuxTransportSslContext
+     * into the NettyServerBuilder config to enable TLS on the server.
+     * @param settings the configured settings.
+     * @param services the gRPC compatible services to be registered with the server.
+     * @param networkService the bind/publish addresses.
+     * @param secureTransportSettingsProvider TLS configuration settings.
+     */
+    public SecureNetty4GrpcServerTransport(
+        Settings settings,
+        List<BindableService> services,
+        NetworkService networkService,
+        SecureAuxTransportSettingsProvider secureTransportSettingsProvider
+    ) {
+        super(settings, services, networkService);
+        this.port = SecureNetty4GrpcServerTransport.SETTING_GRPC_SECURE_PORT.get(settings);
+        this.portSettingKey = SecureNetty4GrpcServerTransport.SETTING_GRPC_SECURE_PORT.getKey();
+        try {
+            JdkSslContext ctxt = getSslContext(settings, secureTransportSettingsProvider);
+            this.addServerConfig((NettyServerBuilder builder) -> builder.sslContext(ctxt));
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to build SslContext for " + SecureNetty4GrpcServerTransport.class.getName(), e);
+        }
+    }
+
+    private JdkSslContext getSslContext(Settings settings, SecureAuxTransportSettingsProvider provider) throws SSLException {
+        SSLContext sslContext = provider.buildSecureAuxServerTransportContext(settings, this).orElseThrow(IllegalArgumentException::new);
+        SecureAuxTransportSettingsProvider.SecureAuxTransportParameters params = provider.parameters()
+            .orElseThrow(IllegalArgumentException::new);
```

Comment:
Would be good to provide default here, wdyt?

```suggestion
            .orElseGet(DefaultParameters::new);
```